### PR TITLE
[CI] Add github action for macos

### PR DESCRIPTION
**Pull request by Jarod42**
52m job https://github.com/Jarod42/codelite/actions/runs/8343196567
spot the issue about SideBarToolBar :) https://github.com/Jarod42/codelite/actions/runs/8339588463/job/22821858878

[edited]: only 25min here :-)

### CODE REVIEW

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    - name: Build and install Codelite
+      run: |
+        mkdir build-release
+        cd build-release
+        cmake .. -DCMAKE_BUILD_TYPE=Release
+        cmake --build . -j $(sysctl -n hw.physicalcpu)
+        cmake --build . --target install
+
+    - name: codelite --version
+      run: codelite --version || exit 0 # codelite --version returns -1
```
Comment (eranif) [inline .github/workflows/macos.yml:50]: Does this work? it should be:

```bash
codelite.app/Contents/MacOS/codelite --version
```

Comment (Jarod42) [reply .github/workflows/macos.yml]: > /Users/runner/work/_temp/02ffa54b-035d-4b75-bef9-f8728bd7b170.sh: line 1: codelite: command not found

Indeed, not.
Changed with your version, waiting for CI.

**Comment by Jarod42** [comment]
> codelite.app/Contents/MacOS/codelite: No such file or directory

That version doesn't work neither

Whole block can be removed (As important think, the build, passes).
Or macos users might check correct command (I don't know if it is path issue, or something similar).

File: .github/workflows/macos.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          repository: wxWidgets/wxWidgets
+          path: wxWidgets
+          submodules: recursive
+    
+    - name: Build and install wxWidgets
+      run: |
+        mkdir wxWidgets/build-release
+        cd wxWidgets/build-release
+        ../configure --enable-shared --enable-monolithic --with-osx_cocoa CXX='clang++ -std=c++11 -stdlib=libc++ -I../src/tiff/libtiff' CC=clang --disable-debug --disable-mediactrl --enable-stl
+        make -j10
```
Comment (Eric-01) [inline .github/workflows/macos.yml:32]: Why hardcode it to 10 jobs?

Also, CodeLite is now built using C++17, no? Is there any reason why wxWidgets is built using an older standard on macOS (I've never owned a Mac)?

Comment (eranif) [reply .github/workflows/macos.yml:32]: @Jarod42 we could probably use

```bash
 make -j$(sysctl -n hw.ncpu) install
``

Comment (Jarod42) [reply .github/workflows/macos.yml:32]: I mostly copy-paste code from https://docs.codelite.org/build/build_wx_widgets/#macos

I don't have MacOS neither.

Created https://github.com/eranif/codelite/pull/3339.

**Pull request merged**